nomad/search_endpoint.go

```diff
@@ -229,3 +491,140 @@ func (s *Search) PrefixSearch(args *structs.SearchRequest, reply *structs.Search
 		}}
 	return s.srv.blockingRPC(&opts)
 }
+
+// FuzzySearch is used to list fuzzy or prefix matches for a given text argument and Context.
+// If the Context is "all", all searchable contexts are searched. If ACLs are enabled,
+// results are limited to policies of the provided ACL token.
+//
+// These types are limited to prefix UUID searching:
+//   Evals, Deployments, ScalingPolicies, Volumes
+//
+// These types are available for fuzzy searching:
+//   Nodes, Namespaces, Jobs, Allocs, Plugins
+//
+// Jobs are a special case that expand into multiple types, and whose return
+// values include Scope which is a descending list of IDs of parent objects,
+// starting with the Namespace. The subtypes of jobs are fuzzy searchable.
+//
+// The Jobs type expands into these sub types:
+//   Jobs, Groups, Services, Tasks, Images, Commands, Classes
+//
+// The results are in descending order starting with strongest match, per Context type.
+func (s *Search) FuzzySearch(args *structs.FuzzySearchRequest, reply *structs.FuzzySearchResponse) error {
+	if done, err := s.srv.forward("Search.FuzzySearch", args, args, reply); done {
```

Comment:
sorry for asking before the PR is fully teed up. I was just curious why it was necessary to forward to the leader. It seems like it'd be nice to just run it on a follower / whichever address was supplied. In the case of an enterprise cluster this would allow people to target a non-voting node

Reply:
That's an interesting idea! We'd need a way to make sure the state store of the follower is at least "reasonably" up to date (e.g. results from a just-joined server would be empty), do you happen to know of a quick / easy way to check that?

Reply:
Oh, I actually forgot about `AllowStale` which is on `QueryOptions` so as long as that's plumbed through and set, `forward` will no-op